### PR TITLE
make unsafe features start with UNSAFE prefix

### PR DESCRIPTION
**Pull request by ramaraochavali**
This PR ensures
- Unsafe features are enabled only by setting a special env var so that we are choosing them consciously
- The test only features are "hidden" so that they do not show up in docs
- Emits a warning during server startup if they are enabled
[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ X] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[X] Does not have any changes that may affect Istio users.

### CODE REVIEW

**Comment by ramaraochavali** [comment]
@howardjohn This is where I landed with my experiments. Felt little bit over engineered because of env package limitations. But PTAL and let me know what you think?

**Comment by hzxuzhonghu** [comment]
What features are safe/unsafe? It is the user's responsibility to distinguish which features they need, I do not feel in the community we should make them explicit

**Comment by ramaraochavali** [comment]
That is a great point. But two examples of such features 

`PILOT_ENABLE_UNSAFE_RUNTIME_ASSERTIONS` - This should never be enabled in production code. If they enable it might panic. It only used for our purporse.

ENABLE_ADMIN_ENDPOINTS - These are only added for tests. Again they are not needed for production usage.

So when we know the feature is unsafe to enable/only for test/debug purpose - we should mark it as unsafe via "UNSAFE_" so that they do not get accidentally enabled in prod code.

**Comment by ramaraochavali** [comment]
@howardjohn @hzxuzhonghu 
BTW, this is just an experiment that I was trying.  If you think this is not wort it - I am happy to drop this PR :-)

**Comment by hzxuzhonghu** [comment]
IMO, this is a little complicated, how about simply adding `UNSAFE` prefix?

**Comment by ramaraochavali** [comment]
Ok. Sure I can add that. Do you want them those appear in docs? I assume yes?

**Comment by ramaraochavali** [comment]
@hzxuzhonghu Fixed. PTAL

File: pilot/pkg/features/pilot.go [hunk 1]
```diff
@@ -417,9 +417,14 @@ var (
...
+		"UNSAFE_PILOT_ENABLE_RUNTIME_ASSERTIONS",
 		false,
 		"If enabled, addition runtime asserts will be performed. "+
 			"These checks are both expensive and panic on failure. As a result, this should be used only for testing.",
 	).Get()
 )
+
+// UnsafeFeaturesEnabled returns true if any unsafe features are enabled.
+func UnsafeFeaturesEnabled() bool {
+	return EnableUnsafeAdminEndpoints && EnableUnsafeAssertions
```
Comment (hzxuzhonghu) [inline pilot/pkg/features/pilot.go:429]: ||

**Comment by hzxuzhonghu** [comment]
I think this lg

File: pilot/pkg/features/pilot.go (hunk 1 above)
Comment (ramaraochavali) [reply pilot/pkg/features/pilot.go:429]: :-). Fix it.

**Comment by ramaraochavali** [comment]
@howardjohn PTAL

**Review by howardjohn — APPROVED** [review]

**Pull request merged**